sphinx/domains/python.py

```diff
@@ -641,6 +641,152 @@ def generate(self, docnames=None):
         return sorted_content, collapse
 
 
+class PythonClassIndex(Index):
```

Comment:
It seems these three index classes are very similar.
How about making a common parent class?
It would be nice to keep code simple :-)